force-app/main/default/classes/UserAdoptionBatch.cls

```diff
@@ -0,0 +1,81 @@
+    
+    global UserAdoptionBatch(Boolean isNewUsers){
+        this.isNewUsers = isNewUsers;
+    }
+
+    // Necessary for the class to be schedulable as well as batchable
+    global UserAdoptionBatch(){        
+    }
+    
+    global void execute(SchedulableContext sc) {
+        UserAdoptionBatch btch = new UserAdoptionBatch(false);
+        Database.executeBatch(btch, BATCH_LIMIT);
+    }
+
+    global Iterable<AggregateResult> start(Database.BatchableContext BC) {       
+        String query = '';
+        // Query and thus batch type depends on the isNewUsers flag.
+        if(!isNewUsers){            
+            if(Test.isRunningTest()){
+                query = 'SELECT CompanyName, Min(Region__c) Region, COUNT(ID) Logins FROM User WHERE Username LIKE \'%@nav.apextest%\' GROUP BY CompanyName';
+            }else{
+                query = 'SELECT CompanyName, Min(Region__c) Region, COUNT(ID) Logins FROM User WHERE LastLoginDate >= ' + String.valueOf(periodDate).replace(' ','T') + 'Z GROUP BY CompanyName';
+            }
+        } else {
+            if(Test.isRunningTest()){
+                query = 'SELECT CompanyName, Min(Region__c) Region, ProfileId, COUNT(ID) Users FROM User WHERE CreatedDate >= ' + String.valueOf(startDate.addDays(1)).replace(' ','T') + 'Z AND CreatedDate <= ' + String.valueOf(endDate.addDays(1)).replace(' ','T') + 'Z GROUP BY CompanyName, ProfileId';
+            }else{
+                query = 'SELECT CompanyName, Min(Region__c) Region, ProfileId, COUNT(ID) Users FROM User WHERE CreatedDate >= ' + String.valueOf(startDate).replace(' ','T') + 'Z AND CreatedDate <= ' + String.valueOf(endDate).replace(' ','T') + 'Z GROUP BY CompanyName, ProfileId';
+            }
+        }
+        return new AggregateResultIterable(query);       
+    }
+
+    global void execute(Database.BatchableContext BC, List<sObject> scope) {
+        List<ReportingData__c> reportingData = new List<ReportingData__c>();
+        Date rD = Date.newInstance(reportDate.year(), reportDate.month(), reportDate.day());
+        for(sObject sObj : scope){
+            AggregateResult ar = (AggregateResult)sObj;
+            ReportingData__c r = new ReportingData__c(
+                CRM_Region__c = (String) ar.get('Region'),                
+                CRM_Date__c = rD,
+                CRM_NavUnit__c = (String) ar.get('CompanyName'));
+            if(!isNewUsers){
+                r.Name = 'Pålogging';
+                r.CRM_Value__c = (Integer) ar.get('Logins');
+                r.CRM_Category__c = 'User Adoption';
+            } else {
+                String profileName = (String) ar.get('ProfileId');
+                r.Name = profileNameById.get(profileName);
+                r.CRM_Value__c = (Integer) ar.get('Users');
+                r.CRM_Category__c = 'New Users';
+            }
+            reportingData.add(r);            
+        }
+        insert reportingData;
```

Comment:
Add a try/catch and some logging around the insert. See here: https://github.com/navikt/crm-hot/blob/ae63138413d49f3b895ebcf0fe1072c15afab59f/force-app/main/sms/classes/HOT_ReminderSMSBatch.cls#L28